### PR TITLE
Update O2RunSim.h

### PR DESCRIPTION
**Pull request by cholmcc**
`fLoaderName` is an object, not a pointer.  Correct access operator from `->` to `.` for `Data` member function.

### CODE REVIEW

**Comment by shahor02** [comment]
@cholmcc the CI error is genuine: 
```
input_line_9:259:58: error: member reference type 'TString *' is a pointer; did you mean to use '->'?
    FairGeoLoader* loader = new FairGeoLoader(fLoaderName.Data(), "Geo Loader");
                                              ~~~~~~~~~~~^
```

the compiler rightfully complains that the fLoaderName is a pointer: https://github.com/alisw/FairRoot/blob/ebbc65e18e4c0165c264e6894f2fd528fb26f445/base/steer/FairRunSim.h#L179

**Comment by cholmcc** [comment]
> the compiler rightfully complains that the fLoaderName is a pointer:

Well

https://github.com/FairRootGroup/FairRoot/commit/925099e3f9e1a3cf92e917da9144027ccc8891df

but since Alice uses an old `FairRoot` it is still a pointer - sigh.    As far as I can tell, there's no way to discover the FairRoot version being compiled against (e.g., a preprocessor macro like `FAIRROOT_VERSION` - it is set in `FairVersion.h` but to the empty string when building from git `master`), so it cannot be made contingent on that.  

Guess this merge must wait until ALICE uses a newer (>=18.8.1) FairRoot.

File: Steer/include/Steer/O2RunSim.h [hunk 1]
```diff
@@ -60,7 +60,7 @@ class O2RunSim : public FairRunSim
     fRootManager->InitSink();
 
     // original FairRunSim follows
-    FairGeoLoader* loader = new FairGeoLoader(fLoaderName->Data(), "Geo Loader");
+    FairGeoLoader* loader = new FairGeoLoader(fLoaderName.Data(), "Geo Loader");
```
Comment (shahor02) [inline Steer/include/Steer/O2RunSim.h]: With `GetGeoModel()->Data()`  I guess it should work both with old and new versions, will give a try

Comment (cholmcc) [reply Steer/include/Steer/O2RunSim.h]: Good idea.

**Comment by sawenzel** [comment]
ALICE is using a well-defined FairRoot version. What is the motivation to use something different than the one we compile against in alidist?

**Comment by cholmcc** [comment]
I really dislike `alidist`.  It's bloated as all <expletive>...</expletive>.  I prefer if I can install a distribution package, since I know that will work with the rest of the system.   For those software packages for which there is no distribution package, I will often go to some length to see if I can make the package myself.   If that becomes to cumbersome, then I will pull down the package myself and built it.    That means I will often have `master` of a package for which ALICE uses some ancient version in `alidist`.   Therefore, I sometimes see inconsistencies between ALICE software and third-party software that you would only discover the hard way through alidist.   So there - because I can :smile:

**Comment by pzhristov** [comment]
> I really dislike `alidist`. It's bloated as all .... I prefer if I can install a distribution package, since I know that will work with the rest of the system. For those software packages for which there is no distribution package, I will often go to some length to see if I can make the package myself. If that becomes to cumbersome, then I will pull down the package myself and built it. That means I will often have `master` of a package for which ALICE uses some ancient version in `alidist`. Therefore, I sometimes see inconsistencies between ALICE software and third-party software that you would only discover the hard way through alidist. So there - because I can 😄

It is very simple - take alibuild and alidist as development packages and update them regularly, then you will have always the consistent set of software. And of course you can have many packages in development mode.

**Comment by cholmcc** [comment]
> It is very simple - take alibuild and alidist as development packages and update them regularly, then you will have 
> always the consistent set of software. And of course you can have many packages in development mode.

I do not agree that it is simple - it is a pain in the ... to set up in any meaningful way outside of CERN - and I'm not the only one who thinks that.  Given that I know a little about the inner workings of `alidist` (or rather `module`), I don't think its a good solution - at least not for someone like myself who knows how to do most stuff :smile:

**Comment by cholmcc** [comment]
> Dear @cholmcc, I will delete every message you write, where you use your specific sense of humour. 
> Thank you for your understanding.

In the message you deleted, I believe I only stated why I do not like `alidist` and why others also dislike `alidist` - how was that in any sense offensive, or would warrant deletion?   Censoring criticism isn't the way to move forward.   If you really think something need to be removed, then I think the best way is to go about is to ask the author to do so, and we can have a discussion. 

Yours,

_Christiam_

**Comment by sawenzel** [comment]
> > Dear @cholmcc, I will delete every message you write, where you use your specific sense of humour.
> > Thank you for your understanding.
> 
> In the message you deleted, I believe I only stated why I do not like `alidist` and why others also dislike `alidist` - how was that in any sense offensive, or would warrant deletion? Censoring criticism isn't the way to move forward. If you really think something need to be removed, then I think the best way is to go about is to ask the author to do so, and we can have a discussion.
> 
> Yours,
> 
> _Christiam_

I think a constructive way forward would be to come to the relevant meetings (computing board, WP3 meeting) where the software packaging approach is discussed and present your (and the anonymous+) worries. This would be way better than ranting about them on the Github platform as side-remarks. In the end, choices about these systems were taken by our organisation some time ago and most people do not have problems following these common workflows (even in disagreement) because standardized approaches are a core foundation of working together efficiently.

**Comment by pzhristov** [comment]
Dear @cholmcc, I have no problem to discuss  alidist and your proposals, but explaining the problems you have with it in medical terms is not appropriate :) You can move all the discussion in the alidist repository or in JIRA, put there your comments and then we review them at the next WP3 meeting. As you know, I appreciate your knowledge and skills, but I disagree with the style of your communication here. Anyway, you are proposing __one line change__ that does not deserve a  full article in the chat :)

**Comment by cholmcc** [comment]
> Dear @cholmcc, I have no problem to discuss alidist and your proposals, but explaining the problems you have with it
> in medical terms is not appropriate :) 

Sorry, I'm not sure what you mean "medical terms"?   Of course, I cannot see what I wrote since you removed it, but I'm fairly confident that all I said was that `alidst` isn't as easy to deal with as you might think, and I'm not alone in that sentiment, and because I know how to things differently, I do that and get rid of the bloat of `alidist`. 

> .. but I disagree with the style of your communication here. 

and I disagree with removing other peoples post  - so there you have it: which is worse: an expletive or censoring?

**Comment by pzhristov** [comment]
@cholmcc Ok, I consider that the alidist discussion here is off-topic and this is my last contribution to it. Especially for you I have opened the discussions in the alidist repository: https://github.com/alisw/alidist/discussions/5322 Fill free to share there all your comments and proposals.

**Comment by pzhristov** [comment]
> @cholmcc Ok, I consider that the alidist discussion here is off-topic and this is my last contribution to it. Especially for you I have opened the discussions in the alidist repository: [alisw/alidist#5322](https://github.com/alisw/alidist/discussions/5322) Fill free to share there all your comments and proposals.

And now I will move the posts there and remove them from this ticket, even if @cholmcc may not like it.

**Comment by cholmcc** [comment]
> And now I will move the posts there and remove them from this ticket, even if @cholmcc may not like it.

As long as you are not _removing_ stuff in a censorious way _and_ leave a clear link - then, by all means, good to move it elsewhere :smile:

**Comment by pzhristov** [comment]
> > And now I will move the posts there and remove them from this ticket, even if @cholmcc may not like it.
> 
> As long as you are not _removing_ stuff in a censorious way _and_ leave a clear link - then, by all means, good to move it elsewhere 😄

I appreciate your opinion, but let me also have my "administrative" view on subject  😄 Now you have a place where to add more information, feedback, proposals, etc. I am going to remove in a moment all the `alidist` discussion from this ticket.

**Pull request closed without merging**